packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -217,7 +217,7 @@ function s(
                     // undefined is for root components, but root components cannot accept slotted content
                     setVMBeingRendered(slotset.owner!);
                     try {
-                        ArrayPush.apply(newChildren, vnode.factory(data.slotData) as VNode[]);
+                        ArrayPush.call(newChildren, vnode.factory(data.slotData, data.key));
```

Comment:
I assume this `data.key` may be undefined, in which case the function will get the default `key`? (E.g. in the function signature: `function (row, key = 4) {}`.)

If so, it may be good to comment that somewhere, because it's not obvious to me unless I look at the generated code.